pkg/datamover/testutils.go

```diff
@@ -0,0 +1,438 @@
+			"endpoint": []byte(testAwsLocationEndpoint),
+		},
+	}
+
+	s3CredsCreated, err := cli.CoreV1().Secrets(namespace).Create(ctx, s3Creds, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+	s3LocationCreated, err := cli.CoreV1().Secrets(namespace).Create(ctx, s3Location, metav1.CreateOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return s3CredsCreated, s3LocationCreated, nil
+}
+
+func createTestKopiaTLSCertificateSecret(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Secret, error) {
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(2019),
+		Subject: pkix.Name{
+			Organization:  []string{"Test Organization"},
+			Country:       []string{"Test Country"},
+			Province:      []string{"Test Province"},
+			Locality:      []string{"Test Locality"},
+			StreetAddress: []string{"Test Street"},
+			PostalCode:    []string{"123456"},
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(0, 0, 1),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+	caPrivKey, err := rsa.GenerateKey(crand.Reader, 4096)
+	if err != nil {
+		return nil, err
+	}
+	caBytes, err := x509.CreateCertificate(crand.Reader, ca, ca, &caPrivKey.PublicKey, caPrivKey)
+	if err != nil {
+		return nil, err
+	}
+
+	caPEM := new(bytes.Buffer)
+	err = pem.Encode(caPEM, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	caPrivKeyPEM := new(bytes.Buffer)
+	err = pem.Encode(caPrivKeyPEM, &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(caPrivKey),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	tls := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-repository-server-tls-cert-",
+		},
+		Type: "kubernetes.io/tls",
+		Data: map[string][]byte{
+			"tls.crt": caPEM.Bytes(),
+			"tls.key": caPrivKeyPEM.Bytes(),
+		},
+	}
+
+	tlsCreated, err := cli.CoreV1().Secrets(namespace).Create(ctx, tls, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return tlsCreated, nil
+}
+
+func createTestKopiaRepositoryServerUserAccessSecret(ctx context.Context, cli kubernetes.Interface, namespace string) (*corev1.Secret, error) {
+	userAccess := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-repository-server-user-access-",
+		},
+		Data: map[string][]byte{
+			defaultKopiaRepositoryHost: []byte(testKopiaRepoServerAdminPassword),
+		},
+	}
+
+	userAccessCreated, err := cli.CoreV1().Secrets(namespace).Create(ctx, userAccess, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return userAccessCreated, nil
+}
+
+func createTestKopiaRepository(location *corev1.Secret, cli kubernetes.Interface, namespace string, pod *corev1.Pod) error {
```

Comment:
This is also exposed as a part of unit tests PR - https://github.com/kanisterio/kanister/pull/1997/files#diff-08634cb22558c4161eee9d22f5c07033f715ecf7b2d3b90aeec0d44dfab92347R471. Can we reuse it?

Reply:
I think the function you mention takes RepoServer CR as an input. But in this test, we are not creating any CR. So I don't think we can use this function.